tracing/tracing.sh

```diff
@@ -31,3 +31,145 @@ function sync_main_kw_file_with_tracing()
   printf '%s' "$main_kw_file_with_tracing" > "${bin}/${kw_path}"
   chmod +x "${bin}/${kw_path}"
 }
+
+# This function injects code that enables tracing in every kw library file (files
+# inside the `src` directory) and installs them. Files pertaining to Bash and Zsh
+# completions (`src/bash_autocomplete.sh` and `src/_kw`), and the `src/VERSION`
+# aren't injected.
+#
+# @input_dir: Path to directory where base library files are stored
+# @output_dir: Path to directory where altered files with tracing will be installed
+function sync_kw_lib_files_with_tracing()
+{
+  local input_dir="$1"
+  local output_dir="$2"
+  local file_with_tracing
+  local parent_directory
+  local all_filepaths
+  local tmp_dir_path
+
+  tmp_dir_path=$(mktemp --directory)
+
+  all_filepaths=$(find "$input_dir" -type f)
+
+  while IFS=$'\n' read -r filepath; do
+    # We shouldn't try to inject code in the completions files nor the `VERSION` file
+    if [[ "$filepath" =~ src\/(_kw|bash_autocomplete\.sh|VERSION) ]]; then
```

Comment:
Maybe you could create a global array named `ignored_files` where you add all file names that should be ignored and make this line more generic by checking the array. I also think this approach will make it easy to maintain the trace option in the future.

Reply:
Done